platform.md

```diff
@@ -7,64 +7,114 @@ A platform orchestrates a lifecycle to make buildpack functionality available to
 Examples of a platform might include:
 
 1. A local CLI tool that uses buildpacks to create OCI images
-2. A plugin for a continuous integration service that uses buildpacks to create OCI images
-3. A cloud application platform that uses buildpacks to build source code before deployment
```

Comment:
I'd prefer to keep the numbering explicit, so that it's easier to find sections without a markdown rendering tool For example, if someone says "see section 1, item 4b," it's difficult to find that item without rendering the document.  Not having to render the document to read it is generally why I prefer markdown to other markups. That said, I'm okay with this change if others disagree with me (and if we make it consistently).

(On that note, I've noticed that many specifications avoid unordered (e.g., bulleted) lists entirely, because it makes referring to individual items difficult. Would be great if we could pick a consistent rule for lists like "always use `1.` -> `a.` -> `i.` and never use explicit numbering")

Reply:
This numbering thing is a problem that I don’t think we have a good handle on yet.  Including the numbers explicitly requires users to always remember to update them whenever a change is made (and even the best can [forget updates](https://github.com/buildpacks/spec/pull/108#pullrequestreview-443457252)).  Also using ordered lists has a tendency to break links as data is inserted and remove ([note](https://github.com/buildpacks/spec/pull/86/commits/4a57490669e7889add1f541b691d580907f6f207) the anchor names in the TOC).

My personal preferences run towards ordered lists for the same reasons you like them, but the downsides of using them are so high that I can’t justify it 😞

Reply:
> if someone says "see section 1, item 4b," it's difficult to find that item without rendering the document

How likely are you to lookup "section 1, item 4b" in the unrendered version though? Given that we are a bit inconsistent right now I wonder if we could bring decide on guidelines and bring all the specs into compliance with them separately from this PR.